dashboard/src/components/common/Tab.tsx

```diff
@@ -14,56 +14,73 @@ export default function Tab({
   selectedIndex,
   onChange = () => {},
   onTabClose = () => {},
+  isLoading,
 }: {
   tabs: { title: string; action?: () => void; className?: string; width?: string; closable?: boolean }[];
   children?: JSX.Element | JSX.Element[];
   defaultIndex?: number;
   selectedIndex?: number;
   onChange?: (index: number) => void;
   onTabClose?: (i: number) => void;
+  isLoading?: boolean;
 }) {
+  const getTabContent = (
+    tab: { title: string; action?: () => void; className?: string; width?: string; closable?: boolean },
+    idx: number,
+  ) => {
+    if (tab.closable && selectedIndex === idx && tabs.length > 2) {
```

Comment:
Nit: If you moved `selectedIndex` into an argument to this function, could the function be pulled out of the component constructor and used as a normal function? The current structure will recreate a closure every time `Tab` is called.